src/topology/algebra/matrix.lean

```diff
@@ -13,31 +13,149 @@ This file is a place to collect topological results about matrices.
+  continuous (det : matrix n n R → R) :=
 begin
-  suffices : ∀ (n : ℕ), continuous (λ A : matrix (fin n) (fin n) k, matrix.det A),
-  { have h : (det : matrix ι ι k → k) = det ∘ reindex (fintype.equiv_fin ι) (fintype.equiv_fin ι),
-    { ext, simp, },
-    rw h,
-    apply (this (fintype.card ι)).comp,
-    exact continuous_pi (λ i, continuous_pi (λ j, continuous_apply_apply _ _)), },
-  intros n,
-  induction n with n ih,
-  { simp_rw coe_det_is_empty,
-    exact continuous_const, },
-  simp_rw det_succ_column_zero,
-  refine continuous_finset_sum _ (λ l _, _),
-  refine (continuous_const.mul (continuous_apply_apply _ _)).mul (ih.comp _),
-  exact continuous_pi (λ i, continuous_pi (λ j, continuous_apply_apply _ _)),
+  show continuous (λ A : matrix n n R, A.det),
+  simp_rw matrix.det_apply,
+  refine continuous_finset_sum _ (λ l _, continuous.const_smul _ _),
+  refine continuous_finset_prod _ (λ l _, continuous_matrix_elem _ _),
 end
+
+lemma continuous_update_column [decidable_eq n] (i : n) :
+  continuous (λ x : matrix m n R × (m → R), x.1.update_column i x.2) :=
+continuous_matrix $ λ j k, begin
+  obtain rfl | h := decidable.eq_or_ne k i,
+  { simp_rw matrix.update_column_self,
+    exact (continuous_apply _).comp continuous_snd },
+  { simp_rw matrix.update_column_ne h,
+    exact (continuous_matrix_elem _ _).comp continuous_fst },
+end
+
+lemma continuous_update_row [decidable_eq m] (i : m) :
+  continuous (λ x : matrix m n R × (n → R), x.1.update_row i x.2) :=
+continuous_matrix $ λ j k, begin
+  obtain rfl | h := decidable.eq_or_ne j i,
+  { simp_rw matrix.update_row_self,
+    exact (continuous_apply _).comp continuous_snd },
+  { simp_rw matrix.update_row_ne h,
+    exact (continuous_matrix_elem _ _).comp continuous_fst },
+end
+
+lemma continuous_cramer [fintype n] [decidable_eq n] [comm_ring R] [topological_ring R]:
+  continuous (λ A : matrix n n R × (n → R), A.1.cramer A.2) :=
+continuous_pi $ λ i, continuous_det.comp $ continuous_update_column _
+
+lemma continuous_adjugate [fintype n] [decidable_eq n] [comm_ring R] [topological_ring R] :
+  continuous (adjugate : matrix n n R → matrix n n R) :=
+continuous_matrix $ λ j k, continuous_det.comp $ begin
+  show continuous (
+    (λ (a : matrix n n R × (n → R)), a.1.update_column k a.2) ∘
+    (λ a, (a, pi.single j 1)) ∘
+    transpose),
+  exact (continuous_update_column k).comp
+    (continuous.comp (continuous_id.prod_mk continuous_const) continuous_transpose),
+end
+
+/-- When `ring.inverse` is continuous on the units (such as in a normed_ring, or a
+`topological_field`), so is `matrix.has_inv`. -/
+lemma continuous_on_inv [fintype n] [decidable_eq n] [comm_ring R] [topological_ring R]
+  (h : continuous_on ring.inverse {r : R | is_unit r }) :
+  continuous_on (has_inv.inv : matrix n n R → matrix n n R) {A | is_unit A.det} :=
+(h.comp continuous_det.continuous_on (set.maps_to_preimage _ _)).smul
+  continuous_adjugate.continuous_on
```

Comment:
I have no idea what the best way to state this is, as I'm not familiar with using `continuous_on` vs `continuous_at` vs `continuous_within_at`.